components/Typography/src/UIFont+MaterialScalable.h

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+#import "MDCFontTextStyle.h"
+
+@interface UIFont (MaterialScalable)
+
+/**
+ An associated scaling curve that can be used to create resized versions of the font based on a
+ UIContentSizeCategory.
+
+ An associated scaling curve that is a dictionary that maps UISizeCategory to fontsize.
+ The dictionary keys MUST include a complete set of size categories, from
+ UIContentSizeCategoryExtraSmall to UIContentSizeCategoryExtraExtraExtraLarge AND all
+ UIContentSizeCategoryAccessibility categories.
+
+ The dictionary values are the desired pointSize stored as a CGFloat wrapped in an NSNumber.
+
+ Generally, clients will use MDCFontScaler to attach particular scaling curves to a font.
+
+ Note that this curve should be immutable as it is retained, not copied.
+ */
+@property(nonatomic, strong, nullable, setter=mdc_setScalingCurve:)
```

Comment:
Please make this `copy`. Unless you have a specific reason why retain is absolutely necessary, it should be copied for safety.

Reply:
This is a cost / benefit decision.

What are the odds that anyone (outside of us) are going to set this property?
What are the odds that this third party will give us a mutable dictionary AND change it after it is set?  (In spite of all the documentation saying "Don't do that")

Reply:
Discussed offline.  Convincing arguments were made.
copy FTW!